tests/OmniSharp.Roslyn.CSharp.Tests/TestSourceGenerator.cs

```diff
@@ -30,19 +30,21 @@ public class TestGeneratorReference : AnalyzerReference
     {
         private readonly bool _isEnabledByDefault;
         private readonly Action<GeneratorExecutionContext> _execute;
+        private readonly ImmutableArray<ISourceGenerator> _generators;
 
         public TestGeneratorReference(Action<GeneratorExecutionContext> execute, [CallerMemberName] string testId = "", bool isEnabledByDefault = true)
         {
             Id = testId;
             _isEnabledByDefault = isEnabledByDefault;
             _execute = execute;
+            _generators = ImmutableArray.Create<ISourceGenerator>(new TestSourceGenerator(_execute));
         }
 
         public override ImmutableArray<ISourceGenerator> GetGenerators(string language)
-            => ImmutableArray.Create<ISourceGenerator>(new TestSourceGenerator(_execute));
+            => _generators;
```

Comment:
This originally caused a test failure due to changes from https://github.com/dotnet/roslyn/pull/64729. The fix is to not create a new generator each time these methods are called, and instead re-use the existing generator in order to maintain the same documentId.